### PR TITLE
Fix environment to work on AI machine

### PR DESCRIPTION
**Pull request by bouchardi**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ RUN apt update && \
     build-essential \
     wget \
     vim \
+    libglm-dev \
```
Comment (bouchardi) [inline Dockerfile]: Missing lib for `diff-gaussian-rasterizer` submodule. Still a mystery to me why it ever worked without it.

File: README.md [hunk 2]
```diff
@@ -39,13 +39,13 @@ For development, we suggest running the Docker conrtainer in interactive mode.
 To do so, you must mount the proper resources. 
 
 ```shell
-nvidia-docker run -it --rm -v $(pwd):/workspace/ gaussian_splatting 
+docker run -it --gpus all --rm -v $(pwd):/workspace/ gaussian_splatting
```
Comment (bouchardi) [inline README.md:42]: `nvidia-docker` is deprecated. `docker --gpus all` replicates the same behavior.

File: README.md [hunk 3]
```diff
@@ -131,14 +131,6 @@ camera poses during training, pass the `--pose-free` flag to training script.
 python scripts/train.py -s <path to dataset>
 ```
 
-### 3. Evaluation
```
Comment (bouchardi) [inline README.md:134]: I never actually ran the evaluation or the metric scripts. Although they could be useful for quality check, in dev mode, the best way to assess the reconstruction quality is to look at the model rendering in the viewer.

File: environment.yml [hunk 4]
```diff
@@ -1,17 +1,296 @@
...
   - defaults
 dependencies:
-  - plyfile=0.8.1
-  - python=3.7.13
-  - pip=22.3.1
-  - opencv 
-  - pytorch=1.12.1
-  - torchvision=0.13.1
-  - tqdm
+  - _libgcc_mutex=0.1=conda_forge
```
Comment (bouchardi) [inline environment.yml:8]: I exported the entire working environment. We may be able to trim it, but at least this way we know it works!

File: README.md [hunk 5]
```diff
@@ -39,13 +39,14 @@ For development, we suggest running the Docker conrtainer in interactive mode.
...
-nvidia-docker run -it --rm -v $(pwd):/workspace/ gaussian_splatting 
+docker run -it --gpus all --rm -v $(pwd):/workspace/ gaussian_splatting 
 ```
 
 ### 4. Activate the Conda Environment.
 
 ```shell
-conda activate gaussian_activate
+conda activate gaussian_splatting
+pip install submodules/simple-knn
```
Comment (bouchardi) [inline README.md]: Installing this package at build time fails, but it succeed inside the container. That's not the cleanest solution, but eh, it works!

File: Dockerfile (hunk 1 above)
Comment (bouchardi) [reply Dockerfile]: ah! You actually don't need this if you initialized your submodules recursively!

File: README.md [hunk 6]
```diff
@@ -39,13 +39,14 @@ For development, we suggest running the Docker conrtainer in interactive mode.
...
-nvidia-docker run -it --rm -v $(pwd):/workspace/ gaussian_splatting 
+docker run -it --gpus all --rm -v $(pwd):/workspace/ gaussian_splatting 
 ```
 
 ### 4. Activate the Conda Environment.
 
 ```shell
-conda activate gaussian_activate
+conda activate gaussian_splatting
+pip install submodules/diff-gaussian-rasterization submodules/simple-knn
```
Comment (bouchardi) [inline README.md:49]: no me gusta, mais j'ai pas trouvé mieux 😿

**Comment by Cadotte** [comment]
Tested the full pipeline on the AI Machine, confirmed everything works.

**Pull request merged**